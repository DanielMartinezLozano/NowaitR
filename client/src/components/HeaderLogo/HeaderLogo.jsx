import { Image, TouchableOpacity } from 'react-native';
import { Header } from 'react-native-elements';
import React from 'react';
import firebase from 'firebase';
import styles from './HeaderLogoStyles';

function HeaderLogo() {
  const renderLogOut = () => (
    <TouchableOpacity testID="signOutButton" onPress={() => firebase.auth().signOut()}>
      <Image
        style={styles.logout}
        source={{ uri: 'https://cdn2.iconfinder.com/data/icons/picons-essentials/57/logout-512.png' }}
      />
    </TouchableOpacity>
  );

  return (
    <Header
      testID="HeaderLogo"
      leftComponent={(
        <Image
          source={{
            uri:
              // eslint-disable-next-line max-len
              'https://trello-attachments.s3.amazonaws.com/5fc4dc9893cb2246bcf25278/5fc4dc9993cb2246bcf252be/956c8087d4b40c7ec17bc4050c39294a/big-horizontal-logo.png',
          }}
          style={styles.logo}
        />
          )}
      rightComponent={() => renderLogOut()}
      containerStyle={styles.container}
      placement="center"
    />
  );
}

export default HeaderLogo;
