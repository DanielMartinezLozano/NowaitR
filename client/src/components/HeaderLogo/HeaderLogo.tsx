import { Image, StyleSheet } from 'react-native'
import { Header } from 'react-native-elements'
import React from 'react'
import stylesType from '../../../styles/stylesType'

export default function HeaderLogo () {
  return (
      <Header
          centerComponent={
              <Image
                  source={{
                    uri:
              // eslint-disable-next-line max-len
              'https://trello-attachments.s3.amazonaws.com/5fc4dc9893cb2246bcf25278/5fc4dc9993cb2246bcf252be/b8740e39ac1a9c32d6142c96dde0b757/nowaitr-row-logo.png'
                  }}
                  style={styles.logo}
              />
      }
          containerStyle={styles.container}
          placement="center"
      />
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: stylesType.yellow,
    elevation: 2,
    height: 100,
    shadowColor: '#000',
    shadowOffset: {
      height: 1,
      width: 0
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41
  },
  logo: {
    height: 200,
    width: 200
  }
})
