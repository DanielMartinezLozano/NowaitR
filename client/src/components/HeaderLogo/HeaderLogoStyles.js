import { StyleSheet } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import stylesType from '../../../styles/stylesType';

const HeaderLogoStyles = StyleSheet.create({
  container: {
    backgroundColor: stylesType.yellow,
    elevation: 2,
    paddingTop: getStatusBarHeight(true),
    paddingBottom: 0,
    height: 80,
    shadowColor: '#000',
    shadowOffset: {
      height: 1,
      width: 0,
    },
    shadowOpacity: 0.2,
    shadowRadius: 1.41,
  },
  logo: {
    height: 28,
    width: 180,
    margin: 0,
    marginTop: 5,
  },
  logout: {
    height: 30,
    width: 30,
    padding: 0,
    marginTop: 5,
    opacity: 0.9,
  },
});

export default HeaderLogoStyles;
