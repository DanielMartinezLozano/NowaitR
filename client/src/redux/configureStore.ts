/* eslint-disable no-undefined */
import { applyMiddleware, compose, createStore } from 'redux'
import thunk from 'redux-thunk'
import rootReducer from './reducers'

export default function configureStore (initialState = {} || undefined) {
  return createStore(
    rootReducer,
    initialState,
    compose(applyMiddleware(thunk))
  )
}
