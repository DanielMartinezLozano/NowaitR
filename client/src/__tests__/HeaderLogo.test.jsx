import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import HeaderLogo from '../components/HeaderLogo/HeaderLogo';

describe('HeaderLogo Component', () => {
  test('should be defined', () => {
    const store = configureStore({});
    store.dispatch = jest.fn();

    const headerLogo = renderer.create(
      <Provider store={store}>
        <HeaderLogo />
      </Provider>,
    );

    expect(headerLogo).toMatchSnapshot();
  });
});
