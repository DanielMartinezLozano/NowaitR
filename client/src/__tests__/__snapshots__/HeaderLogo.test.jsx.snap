// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderLogo Component should be defined 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F4F7BE",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "elevation": 2,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-between",
      "paddingHorizontal": 10,
      "paddingTop": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 1.41,
    }
  }
  testID="headerContainer"
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://trello-attachments.s3.amazonaws.com/5fc4dc9893cb2246bcf25278/5fc4dc9993cb2246bcf252be/ca038c621dfdfd3d99016baa3a91b008/big-horizontal-logo.png",
        }
      }
      style={
        Object {
          "height": 35,
          "width": 200,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cdn2.iconfinder.com/data/icons/picons-essentials/57/logout-512.png",
          }
        }
        style={
          Object {
            "height": 30,
            "marginTop": 5,
            "padding": 0,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
